### DISCUSSION TITLE
Add Litestar to the "ASGI Frameworks" section of the documentation 🙏🏻

### DISCUSSION THREAD

**Discussion by geeshta** [Kludex/uvicorn#2095] [Q&A] [answered]
Hello,
[Litestar 2.0](https://litestar.dev) (formerly known as Starlite) is a newer ASGI framework that comes with quite a few things included - not to the extend of [Django](https://www.djangoproject.com) but definitely less minimalist than [Flask](https://flask.palletsprojects.com/en/2.3.x/) or [FastAPI](https://fastapi.tiangolo.com). It used to depend on [Starlette](https://www.starlette.io) and [Pydantic](https://docs.pydantic.dev/latest/) (like FastAPI) but has since removed them from dependencies (validation is done using [msgspec](https://jcristharif.com/msgspec/) instead).
It integrates well with [SQLAlchemy](https://www.sqlalchemy.org), supports [Jinja](https://jinja.palletsprojects.com/en/3.1.x/) rendering, [HTMX](https://htmx.org), has authentication, CORS and CSRF middlewares built-in.
It uses uvicorn as an optional dependency for the CLI. I find it very nice and have been enjoying exploring and learning it a lot so far.
- [Official Litestar Website](https://litestar.dev)
- [Litestar GitHub](https://github.com/litestar-org/litestar)

**Comment by euri10** [comment]
excellent, a PR is welcomed >_ !

**Reply by Kludex** [reply, marked as answer]
I don't feel very comfortable with this request, since I've been personally mistreated by the creator of that framework.

That said, I'll not overstep here. If you want to get this in, go ahead @euri10 .

**Reply by geeshta** [reply]
@Kludex I understand, there's something going on as the other core developers of the project all stepped down yesterday. 

However to separate the art from the artist I find the project itself really interesting and powerful. I'll submit a PR, should be easy enough

**Reply by Kludex** [reply]
I respect @euri10 a lot, and he helped uvicorn for a long time, so whatever he decides is fine by me.

I, personally, will not approve a PR including it.

**Reply by geeshta** [reply]
@Kludex So there has been some development. The former maintainers of the project have returned and in turn, the creator of the framework has stepped down completely and transferred ownership to one of the former maintainers. He also admitted (on Litestar Discord) that his anger has caused bad blood. 
Anyway he is no longer part of Litestar... will you be now more comfortable with having it on the website?

**Reply by Kludex** [reply]
Honestly, no. Also, please stop harassing me for such a small matter. The creator was not the only being aggressive towards me... I have 3 members from that organization blocked.

On this matter, I don't think I'm being unreasonable here. I don't like being disrespected, and I can't dissociate the individuals from the organization they represent.

Once again, as I said, I'm not going to block it. If @euri10 really wants it, he can merge it.

**Comment by euri10** [comment]
> Honestly, no. Also, please stop harassing me for such a small matter. The creator was not the only being aggressive towards me... I have 3 members from that organization blocked.
> 
> On this matter, I don't think I'm being unreasonable here. I don't like being disrespected, and I can't dissociate the individuals from the organization they represent.
> 
> Once again, as I said, I'm not going to block it. If @euri10 really wants it, he can merge it.

Oo, I need to put back notifications I suppose >_
What happened there ? I never had an issue with anyone there so this comes as surprising to me, care to expand @Kludex ?

**Reply by Kludex** [reply]
I can tell you more on private, but it's just a very aggressive behavior towards me with different people from the organization.

I'll lock this issue. There's no reason to keep the hate going.

Anyway, for this issue, this is enough: https://github.com/encode/uvicorn/discussions/2095#discussioncomment-7011816.

**Comment by euri10** [comment]
on the topic itself, I'll merge it, tag me on it as I'm not very active those days here, I feel there's no reason on my side why I wouldn't: my take is that it is just a list, not an endorsement, I merged other's frameworks I used or not, so I have no issues, it happens I use and like Litestar but that's not relevant to me in this.

I also understand and respect @Kludex point of view.
Now if we could all try to move forward that would be great.
Maintaining libraries/projects is hard, takes time, a lot of time, free time.
I have been in those maintainer's shoes where I threw some RTFM to some users in issues, so I understand that it's hard sometimes to overcome frustration of repeating oneself, it's hard to deal with impatience of users who really want that fix not now but two days ago...
But guess what, it's possible to do better, I've also seen pristine projects where dealing with issues is always made in a positive, constructive way (not to name them I think about trio guys and tiangolo), and I think  we can all benefit from trying to mimic that.

At the end of the day we're all, as maintainers, the users of another's maintainer library.
Sorry for the boomer's motivational speech, you can clearly see from reading this I'm not Tom Cruise in Magnolia, so if you truly want to see a real good example of what it could have been I can only encourage you to watch that masterpiece, and remember: "This Was Not Just A Matter Of Chance."
